python/cudf/cudf/core/series.py

```diff
@@ -5216,3 +5218,7 @@ def isclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
         result_col[equal_nulls] = True
 
     return Series(result_col, index=index)
+
+
+def factorize(obj, na_sentinel=-1):
+    return cudf.Series(obj).factorize(na_sentinel=na_sentinel)
```

Comment:
We should match the signature of Pandas: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.factorize.html

Also needs docstring.

Also, should this live somewhere instead of `Series`.

Reply:
I made some updates here. I don't have a strong preference as to where this code lives, although that said, I didn't see anywhere that made more sense than Series, given that it makes heavy use of series methods. It seems strange to go in `frame.py` as the set of objects that this function acts on excludes DataFrame and includes cupy arrays. It could go in `column.py` although I don't see why that makes more sense than where it is. That leaves somewhere in `utils` as far as I can tell, and I have been trying not to use that area as a place where just everything goes that doesn't have a better home. 

I am totally OK with moving it though for really any reason, I am just not sure where.

Reply:
Pandas has it in an `algorithms.py`: https://github.com/pandas-dev/pandas/blob/v1.1.5/pandas/core/algorithms.py#L518-L728

Maybe we should just follow suite?

Reply:
👍 that works for me.

Reply:
Reorganized this